gst/nnstreamer/include/nnstreamer_plugin_api_filter.h

```diff
@@ -504,6 +504,25 @@ nnstreamer_filter_exit (const char *name);
 extern const GstTensorFilterFramework *
 nnstreamer_filter_find (const char *name);
 
+/**
+ * @brief Get neural network framework name from given model file. This does not guarantee the framework is available on the target device.
+ * @param[in] model_files the prediction model paths
+ * @param[in] num_models the number of model files
+ * @param[in] load_conf flag to load configuration for the priority of framework
+ * @return Possible framework name (NULL if it fails to detect automatically). Caller should free returned value using g_free().
+ */
+extern char *
+nnstreamer_filter_detect_framework (const char **model_files, const unsigned int num_models, const int load_conf);
+
+/**
+ * @brief Check if the given hw is supported by the framework.
+ * @param[in] name The name of filter sub-plugin.
+ * @param[in] hw Backend accelerator hardware.
+ * @return TRUE if given hw is available.
+ */
+extern int
+nnstreamer_filter_check_hw_availability (const char *name, const accl_hw hw);
+
```

Comment:
This is "internal API" for filter subplugins, not API implementations.
You may need to add one more header file, exported for devel, at /gst/nnstreamer/include
OR
you can keep it in the original position; you are migrating API headers only.

Reply:
I will reconsider to migrate API (headers only. or all implementation to API repo)